price-alert-script/price-alert-script.py

```diff
@@ -84,34 +155,41 @@ def send_email(ses_client, sender, recipient, subject, body):
             'Body': {'Text': {'Data': body}}
         }
     )
-
     # Change print to log statement that gets logged to log file
 
     print(f"Email sent! Message ID: {response['MessageId']}")
 
 
-def selectively_send_emails(ses_client):
+def selectively_send_emails(ses_client, boolean_list: list[list]):
     """Selectively sending emails to users if price drops"""
 
-    sender = 'trainee.harvind.grewal@sigmalabs.co.uk'
-    recipient = 'trainee.harvind.grewal@sigmalabs.co.uk'
-    subject = 'Test email'
-    body = 'Body of email'
+    # [TODO]: Rename 'boolean list. This refers to the list that comes out of 'get user emails'.
+    # [TODO]: Work out who the 'sender' should be
+    # [TODO]: Include the value of the discount in a concise, computationally efficient way
 
-    # Your condition for selective sending
-    should_send_email = True  # Replace with your own condition
+    sender = 'trainee.tayla.dawson@sigmalabs.co.uk'
 
-    if should_send_email:
-        send_email(ses_client, sender, recipient, subject, body)
-    else:
-        print("Email not sent based on the condition.")
+    for subscription in boolean_list:
+        if subscription[-1] == True:
+            # recipient = 'trainee.tayla.dawson@sigmalabs.co.uk'  # change this to each user
+            recipient = subscription[1]
+            subject = 'Your item has decreased in price!'
+            body = f'Item {subscription[-2]} has decreased in price.'
+
+            send_email(ses_client, sender, recipient, subject, body)
+        else:
+            print("Email not sent based on the condition.")
 
 
 if __name__ == "__main__":
 
     load_dotenv()
     conn = get_database_connection()
-    print(get_prices_of_latest_pair_of_products(conn))
-    ses_client = create_ses_client()
+    latest_data = get_prices_of_latest_pair_of_products(conn)
+
+    # get_prices_of_latest_pair_of_products(conn)
+    user_product_booleans = get_user_emails(conn)
 
-    selectively_send_emails(ses_client)
+    # compare_latest_two_prices(latest_data, 6)
+    ses_client = create_ses_client()
+    selectively_send_emails(ses_client, user_product_booleans)
```

Comment:
Good. Once finished the final thing would to be remove all commented out code and remove print statements.